### PR TITLE
[OnnxModelLoader] Added API getInputNamesAndTypes

### PR DESCRIPTION
**Pull request by hupreti**
Summary: A facility to read ONNX model inputs updated with onnx-define-symbol before loading the model. The namespace containing utilities related onnx-define-symbol are anonymous and thus cannot be used outside the scope of the file. The API reads a modelfile and updates the input names and input glow type after extracting proper shape info from symbol map.

Documentation: None

Test Plan: Unit Testing

### CODE REVIEW

**Comment by hupreti** [comment]
@jfix71 , Please review these changes. We need this API to so that we can use the information passed as -onnx-define-symbol=<string,value> to get names and types information of model inputs _**before loading the model**_.

File: lib/Importer/ONNXModelLoader.cpp [hunk 1]
```diff
@@ -777,6 +777,55 @@ ONNXModelLoader::getTensorType(const ONNX_NAMESPACE::ValueInfoProto &in) {
...
+    std::vector<std::string>::iterator position =
+        std::find(inputs.begin(), inputs.end(), in.name());
+    if (position != inputs.end()) {
+      inputs.erase(position);
+    }
+  }
+
+  for (const auto &finalIn : inputs) {
+    for (const auto &in : graphDef.input()) {
+      if (finalIn.compare(in.name()) == 0) {
```
Comment (jfix71) [inline lib/Importer/ONNXModelLoader.cpp:808]: nit: early continue to remove a level of indentation:
```suggestion
      if (finalIn.compare(in.name()) != 0) {
        continue;
      }
```

**Review by jfix71 — APPROVED** [review]
@hupreti I'm fine with adding this -- but are you planning on calling it somewhere in the ONNXModelLoader itself, or is this being used privately on your end e.g. inside a loader extension or something?

File: lib/Importer/ONNXModelLoader.cpp (hunk 1 above)
Comment (hupreti) [reply lib/Importer/ONNXModelLoader.cpp]: Done

**Comment by hupreti** [comment]
@jfix71, Yes, right now it is being used at our end after extending the ModelLoader, but this API can be used in loadmodel() API if needed. Kindly help to merge these changes.

**Comment by hupreti** [comment]
Hi @jfix71, Kindly help to review and merge the changes

**Comment by hupreti** [comment]
Hi @jfix71, I have included your suggestions in my [commit](https://github.com/pytorch/glow/pull/5153/commits/dd7af2f923cb9376c979a42faa73ff52ae8e8b31) can please you review the changes and help to merge the PR.

**Review by jfix71 — APPROVED** [review]
Thanks, sorry for the delay.

**Pull request closed without merging**